src/rust/runtime/queue/mod.rs

```diff
@@ -94,35 +99,60 @@ impl<T: IoQueue> IoQueueTable<T> {
         QDesc::from((index as u32) + Self::BASE_QD)
     }
 
+    /// Gets the type of the queue.
+    pub fn get_type(&self, qd: &QDesc) -> Result<QType, Fail> {
+        let index: u32 = match self.get_index(qd) {
+            Some(index) => index,
```

Comment:
`box_queue_ptr` has type of `&&Box<dyn IoQueue>` is that what we want?

Reply:
nope, will fix